framework/source/class/qx/util/StringEscape.js

```diff
@@ -37,26 +37,21 @@ qx.Bootstrap.define("qx.util.StringEscape",
     {
       var entity, result = "";
 
-      for (var i=0, l=str.length; i<l; i++)
-      {
-        var chr = str.charAt(i);
-        var code = chr.charCodeAt(0);
+      for (var codePoint of str) {
+        var code = codePoint.codePointAt(0);
```

Comment:
I'm not very familiar with `codePointAt`. Is `codePointAt` guaranteed to yield the same result as `charAt` for characters other than the characters for which this PR is intended to correct? My concern is that this has an effect that is not backward compatible, on other, traditional characters that people might have in their programs.

Reply:
I found a polyfill for it at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/codePointAt

Reply:
But do we really need this polifyll? Since this is an API, not a  syntax thing? The generator doesnÄt have a problem, and the bowsers all support the API.

Reply:
According to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/codePointAt, IE does not support it, so there's a likelihood of BC problems. Am I mistaken?

Reply:
I don't remember - do we still support IE for v6? But anyways, it's probably a quick fix to include the polyfill following the pattern in
https://github.com/qooxdoo/qooxdoo/blob/master/framework/source/class/qx/bom/client/EcmaScript.js and  https://github.com/qooxdoo/qooxdoo/tree/master/framework/source/class/qx/lang/normalize - although we don't have a dedicated polyfill class for the global object, do we? If we were to add it, it would have to be added as `qx.lang.normalize.GlobalObject` and here: https://github.com/qooxdoo/qooxdoo/blob/97418ba9a62fed5eb00efbbac5190fdde66ed24b/framework/source/class/qx/module/Polyfill.js . Is this correct, @derrell ?

Reply:
Forget my last comment about the global object, we are talking about a method of the `String` object, so no new class is needed. Sorry for the confusion.

Reply:
We should polyfill it.

Reply:
> I don't remember - do we still support IE for v6?

Yes!  There are loads of users with IE11, and I think that officially it's not EOL for a while yet